src/mount.ts

```diff
@@ -20,16 +20,60 @@ export type MountOptions = {
   prepareContainer?: (container: HTMLElement) => void;
 };
 
+export type EnzymeSelector = string | FunctionComponent<any>;
+
+/**
+ * Inspired by DefinitelyTyped's ReactWrapper.
+ * @see https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cebb88fecfa52f854826e216a537867c11b0150e/types/enzyme/index.d.ts
+ */
+export type ComponentWrapper<P = {}, S = {}> = {
+  at(index: number): ComponentWrapper;
+
+  childAt(index: number): ComponentWrapper<any, any>;
+  childAt<P2, S2>(index: number): ComponentWrapper<P2, S2>;
+
+  closest<P2>(component: FunctionComponent<P2>): ComponentWrapper<P2, any>;
+  closest(selector: string): ComponentWrapper<JSX.HTMLAttributes, any>;
+
+  exists(selector?: EnzymeSelector): boolean;
+
+  find<P2>(component: FunctionComponent<P2>): ComponentWrapper<P2, any>;
+  find(selector: string): ComponentWrapper<JSX.HTMLAttributes, any>;
+
+  first(): ComponentWrapper<P, S>;
+  forEach(
+    fn: (wrapper: ComponentWrapper<P, S>, index: number) => any,
+  ): ComponentWrapper<P, S>;
+  getDOMNode(): HTMLElement;
+  hasClass(className: string | RegExp): boolean;
+  last(): ComponentWrapper<P, S>;
+  length: number;
+  map<V>(fn: (wrapper: ComponentWrapper<P, S>, index: number) => V): V[];
+
+  prop<K extends keyof P>(key: K): P[K];
+  prop<T>(key: string): T;
+
+  props(): P;
+  setProps<K extends keyof P>(
+    props: Pick<P, K>,
+    callback?: () => void,
```

Comment:
`setProps` always synchronously re-renders and updates the wrapper, at least when used with Preact, so I don't think this callback is useful for us.

Reply:
True. We don't use the second argument anywhere.